### DISCUSSION TITLE
How to communicate to podman from server1 to podman from server2

### DISCUSSION THREAD

**Discussion by Taymindis** [containers/podman#12798] [Q&A]
I am currently designing a podman architecture, if I have 2 vm system with podman support. 

how can I communicate each other via cni network? Should I either use `network=host` or `published port` approach, or am I missing out of some podman feature ??

Giving an illustration below, How should I clustered the connection? By using which ip interface?

![image](https://user-images.githubusercontent.com/14217761/148745354-9126228c-3f9a-4026-b48b-b59c7e3ac5fa.png)

**Comment by mheon** [comment]
Theoretically, you could use a multi-tenant-capable CNI driver to do this - CNI was originally designed to work with Kubernetes, so it's definitely capable of this sort of work.

However, Podman in general is very much focused on single-node only use, so I can say that I personally have never seen this done before, and I don't think anyone on the Podman team has much in the way of expertise about this.

**Comment by Luap99** [comment]
Yeah likely possible with the right cni plugins but this is not really supported by us. I recommend to just forward the needed ports from the host into the container and then connect to the host ip.

**Comment by TomSweeneyRedHat** [comment]
Just so this can be more easily found in the future, I'm going to move this to the discussions.

**Comment by Taymindis** [comment]
Does it have similar feature like docker which show on this [video](https://youtu.be/Xxhhdo2e-DA?t=182)

It creates network overlay drive to share across the nodes

**Comment by rhatdan** [comment]
No Podman has no support for cross Node networking. Or Cross Node anything.  When  you go to multiple nodes we recommend Kubernetes.

**Reply by Taymindis** [reply]
@rhatdan ,

Thanks for your advice.

However, If podman is not supporting for cross node networking. Does it mean the podman is not scalable? 
  
Alternatively,  
Can I assume my N number of vm actually sharing the `network=host` container to fit the goal?

Will that be an issue or any anti-pattern?

**Comment by rhatdan** [comment]
Not sure what you mean by Podman Scalable.  If Scalable means to you running containers on multiple machines via one Podman command, then no it can not do that and is not designed to do that.  (Thats Kubernets+CRI-Os job).

I am not sure what your goal is.

If what you want is two containers running on separate VMs running within the same network and able to see each other, then Podman can not do that.  @Luap99 @mheon Any other ideas?

**Reply by Taymindis** [reply]
@rhatdan 

You are right. 

however,  I don't need one podman to control all the nodes. But I need at least all the containers from different VMs are communicable. 

If you said `kubernetes`. That will be out of my alternative solution as we have our own VM allocating and There are not cloud based. 


I wil try to use `network=host` to achive the goal above, hope that i am not using far away from podman pattern and design
